### PR TITLE
Make tux grab snail

### LINKED ISSUE #534: Tux Should Be Able To Carry Snail
I often die when attempting to pick up the [snail](https://github.com/SuperTux/data/blob/master/images/creatures/snail/snowsnail.png) after jumping on it. Tux should be able to pick it up like Mr. Iceblock

### PR DESCRIPTION
**Pull request by christ2go**
Enables Tux to carry snails. 

Tux can now carry snails similiar to iceblocks. (The Snail class now also inherits the portable class, and the required methods have been implement.)
Closes  #534

### CODE REVIEW

File: src/badguy/mriceblock.cpp [hunk 1]
```diff
@@ -135,6 +135,7 @@ MrIceBlock::collision(GameObject& object, const CollisionHit& hit)
 HitResponse
 MrIceBlock::collision_player(Player& player, const CollisionHit& hit)
 {
+
```
Comment (tobbi) [inline src/badguy/mriceblock.cpp]: Please remove this empty line again.

File: src/badguy/snail.cpp [hunk 2]
```diff
@@ -81,6 +81,13 @@ Snail::be_flat()
   physic.set_velocity_y(0);
 }
 
+void Snail::be_grabbed()
+{
+  state = STATE_GRABBED;
+  sprite->set_action(dir == LEFT ? "flat-left" : "flat-right", 1);
+
```
Comment (tobbi) [inline src/badguy/snail.cpp]: Empty line.

File: src/badguy/snail.cpp [hunk 3]
```diff
@@ -102,12 +109,16 @@ Snail::can_break() const {
...
 {
+  if(state == STATE_GRABBED)
+    return;
   if(frozen)
   {
     BadGuy::active_update(elapsed_time);
     return;
   }
 
+
```
Comment (tobbi) [inline src/badguy/snail.cpp]: Same for these empty lines, no need for those.

**Comment by tobbi** [comment]
After you're done with the review comments, please squash the commits.

**Comment by tobbi** [comment]
Furthermore, add `closes #534` to your commit message.

File: data [hunk 4]
```diff
@@ -1 +1 @@
-Subproject commit bdc5311b6d7a22c445ddcc80143a5fc21368836b
+Subproject commit 0fccfd981388dc8e9c28e963d0f32a18b6e6f08b
```
Comment (maxteufel) [inline data]: Please open an appropriate pull request on SuperTux/data if needed. Otherwise, I don't know if this is a valid change.

**Comment by maxteufel** [comment]
Next time when using cmake, `-DGENERATE_WRAPPER=OFF` instead of editing the file. As for the commit headlines, keep them as short as possible (e.g. `Snail: allow Tux to carry if flat`). Always keep commit messages in imperative, and add a body detailing the change. Also, squash the commits.

File: data (hunk 4 above)
Comment (christ2go) [inline data]: I didn't change anything in the data repository, how can I reset this?

File: data [hunk 5]
```diff
@@ -1 +1 @@
-Subproject commit bdc5311b6d7a22c445ddcc80143a5fc21368836b
+Subproject commit 28012f3c9d89a8987ad85026d4b7487f38276906
```
Comment (maxteufel) [inline data]: You still haven't given us a reason for this change. If data needs to be changed, add a pull request for data too. If not, don't change this.

**Comment by maxteufel** [comment]
Please resolve the merge conflicts. Also, remove the old commit messages from the new commit message.

File: data (hunk 5 above)
Comment (christ2go) [inline data]: @maxteufel Honestly I don't even know what I changed and don't know how to delete it, I haven't made any changes to the data repository.

**Comment by maxteufel** [comment]
```
Tux now carries snail if it is flat, similiar to MrIceBlock.
Implementation is similiar to MrIceBlocks.
```

Redundant word repetition. Give reasons why we need this feature. How is it similar to Ice Blocks? Explain how it's done in the code base? Did you add a new class or used the Portable class? Stuff like that should be visible by looking at the pull request. Does this cause any potential issues?

File: src/badguy/snail.cpp [hunk 6]
```diff
@@ -270,14 +282,43 @@ Snail::collision_squished(GameObject& object)
       }
       be_kicked();
       break;
-
+    case STATE_GRABBED:
     case STATE_KICKED_DELAY:
       break;
 
+
```
Comment (qwertychouskie) [inline src/badguy/snail.cpp]: Extra line again.

**Comment by maxteufel** [comment]
@christ2go Testing reveals that snails can (now?) be grabbed and kicked when having been burnt.

**Comment by christ2go** [comment]
@maxteufel Can you test it again? I believe I fixed it.

**Comment by maxteufel** [comment]
@christ2go Yes, seems to be fixed! :+1:

**Comment by maxteufel** [comment]
@christ2go: Can you squash the commits?

File: src/badguy/snail.cpp [hunk 7]
```diff
@@ -102,6 +108,8 @@ Snail::can_break() const {
 void
 Snail::active_update(float elapsed_time)
 {
+  if(state == STATE_GRABBED)
+    return;
```
Comment (maxteufel) [inline src/badguy/snail.cpp]: Add a newline after this.

File: src/badguy/snail.cpp (hunk 6 above)
Comment (maxteufel) [inline src/badguy/snail.cpp]: Please follow @qwertychouskie's advise.

File: src/badguy/snail.hpp [hunk 8]
```diff
@@ -43,19 +45,22 @@ class Snail : public WalkingBadguy
   std::string get_display_name() const {
     return _("Snail");
   }
-
+  bool is_portable() const;
+  void ungrab(MovingObject& , Direction dir_);
+  void grab(MovingObject&, const Vector& pos, Direction dir_);
```
Comment (maxteufel) [inline src/badguy/snail.hpp]: Add a newline after this.

File: src/badguy/snail.hpp [hunk 9]
```diff
@@ -43,19 +45,22 @@ class Snail : public WalkingBadguy
...
 protected:
   bool collision_squished(GameObject& object);
   void be_normal(); /**< switch to state STATE_NORMAL */
   void be_flat(); /**< switch to state STATE_FLAT */
   void be_kicked(); /**< switch to state STATE_KICKED_DELAY */
-
+  void be_grabbed();
```
Comment (maxteufel) [inline src/badguy/snail.hpp]: Add a newline after this.

File: src/badguy/snail.hpp [hunk 10]
```diff
@@ -43,19 +45,22 @@ class Snail : public WalkingBadguy
...
 private:
   enum State {
     STATE_NORMAL, /**< walking around */
     STATE_FLAT, /**< flipped upside-down */
     STATE_KICKED_DELAY, /**< short delay before being launched */
-    STATE_KICKED /**< launched */
+    STATE_KICKED, /**< launched */
+    STATE_GRABBED
```
Comment (maxteufel) [inline src/badguy/snail.hpp]: You can add a doxygen comment here (for STATE_GRABBED).

**Pull request merged**